pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -303,7 +257,7 @@ func (os *OpenStack) diskIsAttached(instanceID, volumeID string) (bool, error) {
 		return false, err
 	}
 
-	return instanceID == volume.AttachedServerId, nil
+	return instanceID == volume.Attachments[0].ServerID, nil
```

Comment:
what if volume.Attachments is nil here ? may be you could check the same before.